src/jukebox/components/timers/idle_shutdown_timer.py

```diff
@@ -0,0 +1,131 @@
+# RPi-Jukebox-RFID Version 3
+# Copyright (c) See file LICENSE in project root folder
+
+import time
+import os
+import re
+import logging
+import threading
+import jukebox.plugs as plugin
+
+
+logger = logging.getLogger('jb.timers.idle_shutdown_timer')
+SSH_CHILD_RE = re.compile(r'sshd: [^/].*')
+PATHS = ['shared/settings',
+         'shared/audiofolders']
+
+
+def get_seconds_since_boot():
+    # We may not have a stable clock source when there is no network
+    # connectivity (yet). As we only need to measure the relative time which
+    # has passed, we can just calculate based on the seconds since boot.
+    with open('/proc/uptime') as f:
+        line = f.read()
+    seconds_since_boot, _ = line.split(' ', 1)
+    return float(seconds_since_boot)
+
+
+class IdleShutdownTimer(threading.Thread):
+    """
+    Shuts down the system if no activity is detected.
+    The following activity is covered:
+        - playing music
+        - active SSH sessions
+        - changes of configs or audio content
+
+    Note: This does not use one of the generic timers as there don't seem
+    to be any benefits here. The shutdown timer is kind of special in that it
+    is a timer which is expected *not* to fire most of the time, because some
+    activity would restart it. Using threading.Thread directly allows us to
+    keep using a single, persistent thread.
+    """
+    shutdown_after_seconds: int
+    last_activity: float = 0
+    files_num_entries: int = 0
+    files_latest_mtime: float = 0
+    running: bool = True
+    last_player_status = None
+    SLEEP_INTERVAL_SECONDS: int = 10
+
+    def __init__(self, timeout_seconds):
+        super().__init__(name=__class__.__name__)
+        self.shutdown_after_seconds = timeout_seconds
```

Comment:
I think we need to put a few safe guards here. If the time set in the yaml is very low, let's say between 1 and 60 (seconds), we should have a minimum idle time applied. Imagine someone accidentally puts 2 seconds, the system would immediately shut down after it had started.

Reply:
Good point. I've added logic to limit the enabling to timeouts >=60 seconds.